lib/jekyll-feed/generator.rb

```diff
@@ -20,18 +26,59 @@ def generate(site)
     # We will strip all of this whitespace to minify the template
     MINIFY_REGEX = %r!(?<=>|})\s+!
 
-    # Path to feed from config, or feed.xml for default
-    def feed_path
-      if @site.config["feed"] && @site.config["feed"]["path"]
-        @site.config["feed"]["path"]
+    # Returns the plugin's config or an empty hash if not set
+    def config
+      @site.config["feed"] || {}
+    end
+
+    # Determines the destination path of a given feed
+    #
+    # collection - the name of a collection, e.g., "posts"
+    # category - a category within that collection, e.g., "news"
+    #
+    # Will return "feed.xml", or the config-specified default feed for posts
+    # Will return `/feed/category.xml` for post categories
+    # WIll return `/feed/collection.xml` for other collections
+    # Will return `/feed/collection/category.xml` for other collection categories
+    def feed_path(collection: "posts", category: nil)
+      prefix = collection == "posts" ? "/feed" : "feed/#{collection}"
+      if category
+        "#{prefix}/#{category}.xml"
+      elsif collections[collection]["path"]
```

Comment:
We should do nil checking here in case `collections` is nil or `collections[collection]` is nil